### PR TITLE
Review: Weather app with JS

### PR DESCRIPTION
**Pull request by Milypm**
For this project, I created a web app for a fictional weather site.
It's a single page fully built with JavaScript. Webpack was used for bundling different modules and styles.

You should be able to:

- Search by city
- Search by zip code/country
- Select Celsius or Fahrenheit units

Built with:

- JavaScript
- Webpack
- HTML / CSS
- Git

### CODE REVIEW

File: src/js/api.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      const currentWeather = await responseCurrent.json();
+      data.country = currentWeather.sys.country;
+      const { lat } = currentWeather.coord;
+      const { lon } = currentWeather.coord;
+      const oneCallApi = await fetch(`https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&units=${unitParam}&appid=42f266cd35ff527ebd1a7db4f46e623c`, { mode: 'cors' });
+      const responseOneCall = await oneCallApi.json();
+      getData(responseOneCall);
+      return data;
+    } catch (error) {
+      alert('Error: Oops, we could not find that city name :('); // eslint-disable-line no-alert
```
Comment (rindrajosia) [inline src/js/api.js]: Do not use alert()

File: src/js/api.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      const currentWeather = await responseCurrent.json();
+      data.country = currentWeather.sys.country;
+      const { lat } = currentWeather.coord;
+      const { lon } = currentWeather.coord;
+      const oneCallApi = await fetch(`https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&units=${unitParam}&appid=42f266cd35ff527ebd1a7db4f46e623c`, { mode: 'cors' });
+      const responseOneCall = await oneCallApi.json();
+      getData(responseOneCall);
+      return data;
+    } catch (error) {
+      alert('Error: Oops, we could not find that zip code :('); // eslint-disable-line no-alert
```
Comment (rindrajosia) [inline src/js/api.js]: Do not use alert()

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- Good design :+1:
- Good use of ES6 syntax :100: 

To improve :factory_worker: [REQUIRED]:

- [x] Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store, .vscode) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] Linter has found 4822 problems please fix it as linter helps you to have a good code base quality.
- [x] Do not use `alert()` . It’s ugly, impossible to style, stop code execution and display differently on different browsers. Use custom modal instead. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/alert_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/javascript-module-v-10/166782/path/step/97634928/))
- [x] Please rename `docs` folder to `dist` to follow Webpack project structure.
- [x] It will be nice if your app handle error for async/await properly because I still got an error on the console when I enter a location that doesn't exist.


- [x] The flag that is displayed in your APP doesn't correspond to the right country

Nice to have:

- [ ]  it has a ‘`loading`’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
- [ ]  it fetches the current geolocation of the user when the app is first accessed and fetches the weather from that location

     

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hello @Milypm,

**You have done a commendable job on implementing the changes suggested by @rindrajosia :clap:, and the project should be approved.**

However, there is one issue that must be addressed properly before approval. 

### To improve
- [x] .vscode folder is still showing on the project repo. Kindly add in gitignore and ensure that it is not uploaded in the project repo as suggested by the previous reviewer.
The correct way to do it is `/.vscode, /node_modules` and not `.vscode /`. kindly correct this in gitignore file.

**Kindly make the above changes and ask for another review**

Feel free to reach me on slack or leave a comment while mentioning me and I will  be happy to respond

_Slack: Oloo Moses_

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work.  💻 👀 ✨

## However, there are some changes required 👀 🧰 🔨 🔧
- Please make sure to delete the `.vscode` folder before pushing the changes. The folder is still present in this repository.

### Submit these changes and request a new code review! 😄 👍

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier)🧑‍💻🗨️🧑‍💻
# Happy Coding! 🧠 ⌨️ 🖱️ 🖥️

**Review by bellom — APPROVED** [review]
## Hi @Milypm  :clap: 

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**